app/models/concerns/orchestration/tftp.rb

```diff
@@ -51,37 +50,47 @@ def generate_pxe_template
 
     # work around for ensuring that people can use @host as well, as tftp templates were usually confusing.
     @host = self.host
+
     if build?
-      template = host.provisioning_template({:kind => host.operatingsystem.template_kind})
-      failure_missing_template unless template
-      template_name = template.name
+      begin
```

Comment:
Mind to extract this to `pxe_render(kind)` and the other one to `default_pxe_render`? Just so this method doesn't get that big. So that it looks more or less like this:

```ruby
  def generate_pxe_template(kind)
    #...
    return pxe_render(kind) if build?
    default_pxe_render(kind)
  end

private 
    def pxe_render(kind)
      template = host.provisioning_template({:kind => kind})
      return nil unless template
      unattended_render template
    rescue => e
      failure _("Unable to render %{kind} template '%{name}': %{e}") % { :kind => kind, :name => template.try(:name), :e => e }, e
    end

    def default_pxe_render(kind)
      template_name = "#{kind} default local boot"
      template = ProvisioningTemplate.find_by_name(template_name)
      raise Foreman::Exception.new(N_("template '%s' was not found"), template_name) unless template
      unattended_render template, template_name
    rescue => e
      failure _("Unable to render '%{name}' template: %{e}") % { :name => template_name, :e => e }, e
    end
  end
````